tests/Functional/EnglishHumanizationTest.php

```diff
@@ -33,6 +33,26 @@ public function humanizationProvider(): Generator {
 
 		yield 'Month only' => [ 'XXXX-12-XX', 'December' ];
 		yield 'Day only' => [ 'XXXX-XX-12', '12th' ];
+		
+		// https://github.com/ProfessionalWiki/EDTF/issues/80
+		yield 'Some year (4 digits)' => [ 'XXXX', 'some millennium' ];
+
+		yield 'Some year (1 digit)' => [ 'X', 'some year' ];
+		yield 'Some year (2 digit)' => [ 'XX', 'some decade' ];
+		yield 'Some year (3 digits)' => [ 'XXX', 'some century' ];
+		yield 'Scales (4 digits minus)' => [ '-5XXXX', '5 decem millennia BC' ];
+		yield 'Scales (5 digits)' => [ '5XXXXX', '5 hundreds of thousands' ];
+		yield 'Scales (6 digits)' => [ '5XXXXXX', '5 millions' ];
+		yield 'Scales (7 digits)' => [ '5XXXXXXX', '5 tens of millions' ];
+		yield 'Scales (8 digits)' => [ '5XXXXXXXX', '5 hundreds of millions' ];
+		yield 'Scales (9 digits)' => [ '5XXXXXXXXX', '5 billions' ];
+		yield 'Scales (10 digits)' => [ '5XXXXXXXXXX', '5 tens of billions' ];
+		yield 'Scales (11 digits)' => [ '5XXXXXXXXXXX', '5 hundreds of billions' ];
```

Comment:
"5 decem millennia"? What does `decem` mean here?

"5 hundreds of thousands" is just an amount; should it not also have "years" in there somewhere? Is this even a valid EDTF date to begin with? @mzeinstra

Reply:
yes, sometimes the word "year" is missing, I have still to add it where necessary

Reply:
> "5 decem millennia"? What does `decem` mean here?
> 
"Decem millennium" theoretically means "Ten thousand years" it could be also used the expression "tens of thousands"

Reply:
tens of thousands is indeed better, but we can also leave that up to the translators. 

Cosmology seems to me a valid border of amounts of years that we humanise. Maybe a fallback to just presenting the years in numbers?

The most years I can find in the sciences is the half-life of Xenon, which is 18 billion trillion years :)

Reply:
As we should not forget the objective here, that is to make these edtf-string human readable. If 99% of the use cases can be captured than that is ok with me.

Reply:
Previously I also posted this link

https://asistdl.onlinelibrary.wiley.com/doi/epdf/10.1002/pra2.552

in the element.io SemanticMediawiki group because somebody asked about the support for "geological and historical eras".

Regarding the following

> Cosmology seems to me a valid border of amounts of years that we humanise. Maybe a fallback to just presenting the years in numbers?

in my opinion if the user registers a date with unspecified digits and the library outputs a human readable format, and additionally computes correctly comparisons  and intervals, that's really useful.

Still in my opinion the approach to present a year with mixed unspecified digits using numbers for the specified digits and literals for the unspecified (scaled) part, seems also formally rigorous. So using the example above -5XXXXXXX is not to be precise "50 millions years BC" but "5 tens of millions years BC" (in the output the number represents the specified digits, and the literal part the unspecified digits)

So what about, when this is more refined, to publish a demo (which could query the library using Ajax) and to propose this approach to the authors of the format, the Washington' Library of Congress, as far as I know ?

Reply:
Afaik there is no standard for the humanisation of EDTF, humanised examples are used on the library of congres website but they are not the owner of the standard.

So your best efforts here will de facto help set the humanisation standard.

@mnhn-paul  you might be able to provide us with a standard of humanising eras in EDTF. Eg. What would the date: -5XXXXXXX be humanised to? 

Paul is from our museum of natural history and might have more insight into this.